server/world/world.go

```diff
@@ -672,6 +676,90 @@ func (w *World) enableTimeCycle(v bool) {
 	w.set.TimeCycle = v
 }
 
+// StopWeatherCycle disables weather of the World.
+func (w *World) StopWeatherCycle() {
+	if w == nil {
+		return
+	}
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.set.WeatherCycle = false
+}
+
+// StartWeatherCycle enables weather of the World.
+func (w *World) StartWeatherCycle() {
+	if w == nil {
+		return
+	}
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.set.WeatherCycle = true
+}
+
+// setRainTime sets the rain time.
+func (w *World) setRainTime(new int) {
+	if w == nil {
+		return
+	}
+	w.set.RainTime = int32(new)
+}
+
+// setRainLevel sets the rain level.
+func (w *World) setRainLevel(new int) {
+	if w == nil {
+		return
+	}
+	w.set.RainLevel = float32(new)
+}
+
+// setThunderTime sets the thunder time.
+func (w *World) setThunderTime(new int) {
+	if w == nil {
+		return
+	}
+	w.set.ThunderTime = int32(new)
+}
+
+// setThunderLevel sets the thunder level.
+func (w *World) setThunderLevel(new int) {
```

Comment:
Any reason this is an int while the ThunderLevel field is a float32? I have no idea what value this should hold?

Reply:
If I recall correctly, the values used by vanilla is always an int so that's why I did that. And I don't think floats have a use for it, not sure though.

Reply:
I would still probably either make it a bool or a float32. If it's not used as a float32 and always has either 0 or 1, it makes more sense to make it a bool.